core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -253,7 +253,7 @@ private DocumentMapper merge(DocumentMapper mapper, boolean updateAllTypes) {
                 // simulate first
                 MergeResult result = oldMapper.merge(mapper.mapping(), true, updateAllTypes);
                 if (result.hasConflicts()) {
-                    throw new MergeMappingException(result.buildConflicts());
+                    throw new IllegalArgumentException("Merge failed with failures {" + Arrays.toString(result.buildConflicts()) + "}");
```

Comment:
But that is not equivalent? Arrays.toString is a static method, and different than result.buildConflicts().toString()